src/mercury_engine_data_structures/formats/collision.py

```diff
@@ -91,26 +95,102 @@
 )
 
 
+class PolyData:
+    def __init__(self, raw: Container):
+        self._raw = raw
+
+    @property
+    def num_points(self) -> int:
+        return self._raw.num_points
+
+    @num_points.setter
+    def num_points(self, value: int) -> None:
+        self._raw.num_points = value
+
+    @property
+    def boundings(self) -> Vec4:
+        return self._raw.boundings
+
+    @boundings.setter
+    def boundings(self, value: Vec4) -> None:
+        self._raw.boundings = value
+
+    def get_point(self, point_idx: int) -> PointData:
+        return PointData(self._raw.points[point_idx])
+
+    def add_point(self, position: Vec2, idx: int = 0) -> None:
+        """
+        Adds a new point by copying an existing point and inserting it at a specified index
+        param position: the x,y position of the new point
+        param idx: the index the new point will be placed in the poly
+        """
+        new_point = copy.deepcopy(self.get_point(0))
+        new_point.position = position
+        self._raw.points.insert(idx, new_point)
+        self.num_points += 1
+
+
+class PointData:
+    def __init__(self, raw: Container):
+        self._raw = raw
+
+    @property
+    def position(self) -> Vec2:
+        return self._raw.position
+
+    @position.setter
+    def position(self, value: Vec2) -> None:
+        self._raw.position = value
+
+    @property
+    def material_attribute(self) -> int:
+        return self._raw.material_attribute
+
+    @material_attribute.setter
+    def material_attribute(self, value: int = 1) -> None:
+        self._raw.material_attribute = value
+
+
 class CollisionEntry:
     def __init__(self, raw: Container):
         self._raw = raw
 
-    def get_data(self) -> Container:
-        """Returns all data of collision/collision_camera/logic shape"""
+    @property
+    def data(self) -> dict[Vec3, list[dict], Vec4, list[dict]]:
         return self._raw.data
 
-    def get_poly(self, poly_idx: int):
-        """Returns all data associated with a poly (points, boundings)"""
-        return self.get_data().polys[poly_idx]
+    @data.setter
+    def data(self, value: dict[Vec3, list[dict], Vec4, list[dict]]) -> None:
+        self._raw.data = value
+
+    @property
+    def position(self) -> Vec3:
+        return self.data.position
+
+    @position.setter
+    def position(self, value: Vec3) -> None:
+        self.data.position = value
+
+    @property
+    def total_boundings(self) -> Vec4:
+        return self.data.total_boundings
+
+    @total_boundings.setter
+    def total_boundings(self, value: Vec4) -> None:
+        self.data.total_boundings = value
+
+    @property
+    def polys(self) -> list[dict]:
+        return self.data.polys
```

Comment:
do you need this property for something in particular? we really shouldn't be exposing lists of containers if we can avoid it. honestly this is enough of a problem that I wonder if I should make some new type of list that can shadow another list to solve stuff like this